be/src/exprs/agg/factory/aggregate_factory.cpp

```diff
@@ -134,6 +134,12 @@ static const AggregateFunction* get_function(const std::string& name, LogicalTyp
         }
     }
 
+    if (func_version > 6) {
+        if (name == "group_concat") {
+            func_name = "group_concat2";
```

Comment:
the performance will get worse when has none orderby?

Reply:
The main difference lay at the intermediate results, previous way V1  just concat all strings per group, but the new way V2 store intermediate strings in struct{array[]}, with extra array's offsets costs, one offset per group. So the cost may be not large if group is not large, otherwise not.